ingests/un_sdg.py

```diff
@@ -148,5 +186,71 @@ def download_file(
     return content
 
 
+def attributes_description() -> Dict[Any, Any]:
+    goal_codes = get_goal_codes()
+    a = []
+    for goal in goal_codes:
+        url = f"{BASE_URL}/v1/sdg/Goal/{goal}/Attributes"
+        res = requests.get(url)
+        assert res.ok
+        attr = res.json()
+        for att in attr:
+            for code in att["codes"]:
+                a.append(
+                    {
+                        "code": code["code"],
+                        "description": code["description"],
+                    }
+                )
+    att_dict = pd.DataFrame(a).drop_duplicates().set_index("code").squeeze().to_dict()
+    att_dict["PERCENT"] = "%"
+    return att_dict
+
+
+def dimensions_description() -> dict:
+    goal_codes = get_goal_codes()
+    d = []
+    for goal in goal_codes:
+        url = f"{BASE_URL}/v1/sdg/Goal/{goal}/Dimensions"
+        # These should be retrieved from walden not the live SDG api as the live api is updated
+        res = requests.get(url)
+        assert res.ok
+        dims = res.json()
+        for dim in dims:
+            for code in dim["codes"]:
+                d.append(
+                    {
+                        "id": dim["id"],
+                        "code": code["code"],
+                        "description": code["description"],
+                    }
+                )
+    # Making a nested dictionary of the dimensions - probably could be done in the loop but I'm not sure how
+    dim_dict = defaultdict(dict)
+    for dimen in d:
+        dim_dict[dimen["id"]][dimen["code"]] = dimen["description"]
+
+    nan_dict = defaultdict(dict)
```

Comment:
This could be written with comprehensions as
```python
{key: {np.nan: ""} for key in dim_dict.keys()}
```
although your approach is valid too (and perhaps even more readable?)

Reply:
Thank you! 😄